.github/workflows/test.yml

```diff
@@ -13,8 +13,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [12.x, 14.x, 16.x, 18.x]
-        prettier: [2.1, 2.2, 2.4, latest]
+        node: [14.x, 16.x, 18.x, 20.x]
+        prettier: [3.0]
```

Comment:
This is not acceptable, we can't remove compatibility of old version of prettier

Reply:
I guess it would be hard to maintain both sync and async version of plugin. `prettier-plugin-jsdoc@0.4` is stable enough to work with `prettier@2`, so maybe we can release a new major version `1.0.0` that will support only `prettier@3` and add to README something like:


> This plugin is compatible with `prettier@3.x`, but if you want to use this plugin with `prettier@2.x`, then you can install it with `npm i prettier-plugin-jsdoc@prettier-v2`

---

Prettier v2 will no longer receive updates and the plugin ecosystem is already moving towards prettier v3 adoption

The official plugins already dropped `prettier@2` support:
- `@prettier/plugin-ruby` - https://github.com/prettier/plugin-ruby/commit/47a68f24437c10ed42bc73931676e02f73143045
- `@prettier/plugin-xml` - https://github.com/prettier/plugin-xml/commit/8a01168cfdc69d990421f1311aa8447d82a0d8c4

---

Also migration to ESM will allow us not to include `mdast-util-from-markdown` to the bundle and significantly reduce package size. Otherwise if we will include both sync and async versions, the package size will only grow